mlflow/pyfunc/__init__.py

```diff
@@ -1484,7 +1494,10 @@ def batch_predict_fn(pdf, params=None):
                 return client.invoke(pdf).get_predictions()
 
         elif env_manager == _EnvManager.LOCAL:
-            if should_use_spark_to_broadcast_file:
+            if is_spark_connect_enabled and not is_in_databricks_runtime():
+                with tempfile.TemporaryDirectory() as tmpdir:
+                    loaded_model = mlflow.pyfunc.load_model(model_uri, dst_path=tmpdir)
```

Comment:
- Create a temporary directory path.
- Download the model there.
- If the model already exists, skip the download.